### ISSUE TITLE
Adjust volume of ballcatcher sound effect

### ISSUE THREAD

**Issue by Deconimus** [mwpenny/portal64-still-alive#51]
The ballcatcher sound effect is excessively loud when standing too close. This is especially noticeable when wearing headphones.

I'm not sure if it's just an issue of the effect volume being too high. Compared to the PC version it sounds quite distorted which could also cause this (possibly due to downsampling). Some sound effects already employ additional filters like eq, perhaps this could help here. 

Volume scaling of sounds over distance overall could possibly also benefit from some improvements. In this case here, something like clamping to a maximum value could work, so that sound effects won't get any louder starting from a certain minimum distance.

Comparison between Portal 64 and Portal PC volumes (caution: it gets pretty loud):

https://github.com/mwpenny/portal64-still-alive/assets/2451901/840b51f4-bd42-4acc-97c5-7feb048cada0

https://github.com/mwpenny/portal64-still-alive/assets/2451901/a27058cc-7514-49fb-a931-e8846e07f140

**Comment by mwpenny** [comment]
The current volume [is 5](https://github.com/mwpenny/portal64-still-alive/blob/e81fa6f28a4df9f5e7a18249341e4f5e6b3c3195/src/scene/ball_catcher.c#L110). 3D sounds are limited to volume=1 when output. Greater values just affect attenuation. Other sounds use volumes above 1, but 5 does seem excessive.

I imagine the cause of what you are describing is lack of attenuation. The sound is not downsampled.

With headphones, I notice a difference, but not a huge one. I compared to the original game and sound file (`alyx_stunner1.wav`). I've halved the volume to 2.5. It sounds okay to me, but let me know what you think as well.

**Comment by Deconimus** [comment]
> I imagine the cause of what you are describing is lack of attenuation. The sound is not downsampled.

Yes, attenuation is the word I was looking for when talking about "scaling volume over distance". For some reason I thought all sounds were exported in 22.05khz, but the .sox file says otherwise. It's probably just the volume then.

> With headphones, I notice a difference, but not a huge one. I compared to the original game and sound file (`alyx_stunner1.wav`). I've halved the volume to 2.5. It sounds okay to me, but let me know what you think as well.

It sounds fine in most cases now. However, this doesn't fix the high volume when standing right next to it (as shown in the second clip of the first video). As you've described, only attenuation will be affected by volume values > 1. 

I think it's fine that the ballcatcher can be heard from far away, as it's an important gameplay cue. But the maximum volume when not attenuated should perhaps be more limited. The same could be true for other sounds too (like the energy ball "bounce" and "sing"). I guess it's most noticeable here because of the perceived loudness of the `alyx_stunner1.wav` sample.
